### DISCUSSION TITLE
Decorator vs Class-based Namespaces

### DISCUSSION THREAD

**Discussion by ixxie** [miguelgrinberg/python-socketio#903] [Q&A] [answered]
Its a bit unclear from the documentation what are the pros and cons of using decorator-based vs using class-based namespaces; I would love to hear what usecases and contexts warrant one over the other, if any.

**Comment by miguelgrinberg** [comment, marked as answer]
It's just a matter of preference, really, because both methods are equivalent.

**Reply by ixxie** [reply]
Thanks @miguelgrinberg

**Comment by ixxie** [comment]
@miguelgrinberg - a followup question - is it correct that in the decorator approach, registration occurs by the prefix of the decorator, i.e. if I want to attach two events to different servers I would do something like:
```
sio1 = socketio.AsyncServer()
sio2 = socketio.AsyncServer()

@sio1.event
def my_first_event(sid, data):
    pass

@sio2.event
def my_second_event(sid, data):
    pass
```

**Reply by miguelgrinberg** [reply]
Yes. Each event will be registered only with the server from which you obtained the decorator.
